### PR TITLE
magit-log-propertize-keywords: Fix regexp search

### PR DESCRIPTION
**Pull request by basil-conto**
#### TL;DR

The `magit-keyword` face is currently applied to keywords in log messages greedily, e.g. the entire message `"[EMACS] Rebind C-]"` receives the `magit-keyword` face due to the trailing `]`.

#### ChangeLog

- Make unused grouping construct shy.
- Avoid applying `magit-keyword-squash` face to whitespace.
- Hoist `magit-log-highlight-keywords` guard out of loop.
- Make keyword regexp non-greedy so as to handle cases like `"[EMACS] Rebind C-]"`.
- Avoid meaningless quoting of `?\]` inside regexp, as per advice in [`(elisp) Regexp Special`](https://www.gnu.org/software/emacs/manual/html_node/elisp/Regexp-Special.html).  This is because backslash quoting does not remove special meaning of `?-` and `?\]` in regexps.

### CODE REVIEW

**Comment by tarsius** [comment]
`\\[` ... `]` *feels* unbalanced. Merging anyway because technically you are right.

**Comment by basil-conto** [comment]
Thanks!
